46.md

```diff
@@ -45,15 +45,25 @@ _remote-signer_ provides connection token in the form:
 bunker://<remote-signer-pubkey>?relay=<wss://relay-to-connect-on>&relay=<wss://another-relay-to-connect-on>&secret=<optional-secret-value>
 ```
 
-_user_ passes this token to _client_, which then sends `connect` request to _remote-signer_ via the specified relays. Optional secret can be used for single successfully established connection only, _remote-signer_ SHOULD ignore new attempts to establish connection with old secret. 
+_user_ passes this token to _client_, which then sends `connect` request to _remote-signer_ via the specified relays. Optional secret can be used for single successfully established connection only, _remote-signer_ SHOULD ignore new attempts to establish connection with old secret.
 
 ### Direct connection initiated by the _client_
 
-_client_ provides a connection token in the form: 
+_client_ provides a connection token using `nostrconnect://` as the protocol, and `client-pubkey` as the origin. Additional information should be passed as query parameters:
+
+- `relay` (required) - one or more relay urls on which the _client_ is listening for responses from the _remote-signer_.
+- `secret` (required) - a short random string that the _remote-signer_ should return as the `result` field of its response.
+- `perms` (optional) - a comma-separated list of permissions the _client_ is requesting be approved by the _remote-signer_
+- `name` (optional) - the name of the _client_ application
+- `url` (optional) - the canonical url of the _client_ application
+- `image` (optional) - a small image representing the _client_ application
```

Comment:
```suggestion
- `image` (optional) - the url of a small image representing the _client_ application
```

I assume we're not expecting clients to pass an entire image blob in binary in the `nostrconnect://` string.

Reply:
I personally wouldn't, but if people want to try they can deal with the consequences of a silly design decision. I don't think we need to bloat the spec with it.

Reply:
I think we should specify best practices for the sake of interoperability.  If a signer expects a URL, for instance, and a reckless client dev misunderstands the spec and passes a binary blob instead, the signer is stuck having to deal with this bad decision.  We could forestall some such implementation problems by encouraging best practices in the spec.

Reply:
It won't hurt the signer unless the blob is so big that the signer crashes. The user on the other hand has to deal with a massive string or impossible QR code. This is at worst a UX thing that clients need to own, but I highly doubt it will ever happen.

Reply:
Again, though, why take the risk?  Let's create a "pit of success" for client devs to fall into by identifying best practices in the spec.  If including a whole image blob in the connection string will make an impossible QR code, then let's not let client devs shoot themselves in the foot that way.

Reply:
Specs should be clear and not burdened by every possible implementation detail we can think of. There is a never ending list of things we could recommend against. Feel free to open another pr, but I'm not going to add it to this one.